### ISSUE TITLE
Warning: Your fping binary doesn't support source address setting (-S)

### ISSUE THREAD

**Issue by badbread** [oetiker/SmokePing#161]
Recently migrated my smokeping installation from Ubuntu 16 to 18 and for some reason I am now getting this error when running 'smokeping --debug'

FPingTwo: WARNING: your fping binary doesn't support source address setting (-S), I will ignore any sourceaddress configurations

fping -v
fping: Version 4.0

I can't figure out how to get it working again, worked fine on my old installation.

**Comment by lelutin** [comment]
hmm ubuntu 18.04 (I'm presuming this is the version you went for..) still has smokeping 2.6.11, but even then there was a line of code that was checking the help output from fping to verify presence of -S in order to enable it only if it's present..

could you confirm which version of ubuntu and wich version of smokeping you're running?

also, could you include configuration for that FPingTwo probe?

finally, include the output of `smokeping --debug`

**Comment by lelutin** [comment]
Hi there! someone on the mailing list figured this one out and just submitted a fix for this:

in FPing.pm, the line that checks for availability of the `-S` option need to be changed to the following:

~~~
$self->{enable}{S} = (`$binary -h 2>&1` =~ /\s-S,?\s/);
~~~

(e.g. in 4.x+ there is now a comma after the -S in the output for the help text)

**Comment by oetiker** [comment]
can you do a PR ?

**Comment by lelutin** [comment]
> can you do a PR ?

I've asked Mike, who reported this on the list, if they could make a pull request in order to give them a chance.

I'll wait for about a week, and if there were no responses or no PRs yet, I'll build one then

**Comment by lelutin** [comment]
Mike replied on the list and asked me to submit the patch.. so I took a look into this and actually I see that it was already fixed in commit:

"72ed157 fing compatibility update. fix for #90"

this was merged to master just before release of smokeping 2.7.0, so @badbread  if
you upgrade to that version or later you'll get the fix.

Unfortunately I see[0] that ubuntu 10.10 still had smokeping 2.6.11. 2.7.3 only made it to "disco" (e.g. 19.04) -- which is actually to be expected since I've packaged 2.7.3 in debian only quite recently.

[0]:https://packages.ubuntu.com/search?keywords=smokeping&searchon=names&suite=all&section=all

I *think* you might be able to install the package from "disco" in your 18.04 host and should just work. all of the javascript library dependencies are the same version in "bionic" and "disco".
